extensions/positron-python/src/client/positron/extension.ts

```diff
@@ -71,6 +74,146 @@ export async function activatePositron(serviceContainer: IServiceContainer): Pro
+                await runAppApi.runApplication({
+                    name: 'Dash',
+                    getTerminalOptions(runtime, document, port, urlPrefix) {
+                        const env: RunAppTerminalOptions['env'] = {};
+                        if (port) {
+                            env.DASH_PORT = port;
+                        }
+                        if (urlPrefix) {
+                            env.DASH_URL_PREFIX = urlPrefix;
+                        }
+                        return {
+                            commandLine: [runtime.runtimePath, document.uri.fsPath].join(' '),
+                            env,
+                        };
+                    },
+                });
+            }),
+
+            vscode.commands.registerCommand('python.runFastAPIApp', async () => {
+                const runAppApi = await getPositronRunAppApi();
+                await runAppApi.runApplication({
+                    name: 'FastAPI',
+                    async getTerminalOptions(runtime, document, port, urlPrefix) {
+                        const appName = await getAppName(document, 'FastAPI');
+                        if (!appName) {
+                            return undefined;
+                        }
+                        const args = [
+                            runtime.runtimePath,
+                            '-m',
+                            'uvicorn',
+                            `${pathToModule(document.uri.fsPath)}:${appName}`,
+                        ];
+                        if (port) {
+                            args.push('--port', port);
+                        }
+                        if (urlPrefix) {
+                            args.push('--root-path', urlPrefix);
+                        }
+                        return { commandLine: args.join(' ') };
+                    },
+                    urlPath: '/docs',
+                });
+            }),
+
+            vscode.commands.registerCommand('python.runFlaskApp', async () => {
+                const runAppApi = await getPositronRunAppApi();
+                await runAppApi.runApplication({
+                    name: 'Flask',
+                    async getTerminalOptions(runtime, document, port, urlPrefix) {
+                        const appName = await getAppName(document, 'Flask');
+                        if (!appName) {
+                            return undefined;
+                        }
+                        const env: RunAppTerminalOptions['env'] = {};
+                        if (port) {
+                            env.SCRIPT_NAME = urlPrefix;
+                        }
+                        return {
+                            commandLine: [
+                                runtime.runtimePath,
+                                '-m',
+                                'flask',
+                                '--app',
+                                `${pathToModule(document.uri.fsPath)}:${appName}`,
```

Comment:
same here, do we need the `:appname`? [it looks like we can use filename, just drop the `.py`](https://flask.palletsprojects.com/en/3.0.x/cli/#application-discovery)

Reply:
Nice find! Thankfully the flask CLI ships with flask so should always be installed.